app/services/concerns/xccdf/rule_group_relationships.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving RuleGroupRelationships
+  module RuleGroupRelationships
+    extend ActiveSupport::Concern
+    included do
+      def save_rule_group_relationships
+        @op_rules_and_rule_groups = @op_rule_groups + @op_rules
+
+        ::RuleGroupRelationship.import!(
+          rule_group_relationships.select(&:new_record?), ignore: true
+        )
+
+        rgr_links_to_remove(::RuleGroupRelationship.all).delete_all
```

Comment:
You definitely won't need `all` of it, that's dangerous :see_no_evil: we will only need the `RuleGroupRelationship` entities for this benchmark. The code copied from the `ProfileRules` is actually bad and it should be fixed by limiting the scope for the given benchmark. You'll need to join your model with both rules and groups and check for the right benchmark.

So you'd need a base like:
```ruby
::RuleGroupRelationship.joins(left: :benchmark).where(benchmark: { ref_id: ... })
```

Reply:
I looked into this more closely and your implementation might be correct. I'll do some testing.

Reply:
Oh right, I forgot about the `:benchmark` part! That would be needed for sure. But yeah, I looked into it more and it should be correct but I was to going to create a test for it to make sure(ran out of time yesterday).